### PR TITLE
virt-controller, template: Remove Passt binding SELinux adjustment

### PR DESCRIPTION
**Pull request by EdDev**
### What this PR does

The network Passt binding does not require any special SELinux tweaks to operate now.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
```release-note
NONE
```

### CODE REVIEW

**Comment by acardace** [comment]
@xpivarc @jean-edouard Passt was our last user of the `virt_launcher.process` custom SELinux type, what about if we just remove it completely now?

**Comment by acardace** [comment]
/approve

Thanks for the PR @EdDev, I think the complete removal of the custom policy can be anyway done in a separate PR.

File: pkg/virt-controller/services/template.go [hunk 1]
```diff
@@ -1276,14 +1276,7 @@ func wrapGuestAgentPingWithVirtProbe(vmi *v1.VirtualMachineInstance, probe *k8sv
 	return
 }
 
-func alignPodMultiCategorySecurity(pod *k8sv1.Pod, vmi *v1.VirtualMachineInstance, selinuxType string, dockerSELinuxMCSWorkaround bool, customPolicyDisabled bool) {
-	if selinuxType == "" {
```
[the remark this answers is not in the document]
Comment (EdDev) [reply pkg/virt-controller/services/template.go:1280]: I do not have a confirmed explanation on why it works.
I can confirm it works without it on our e2e tests.

I can guess that it started working once we operate passt through libvirt.

@sbrivio-rh , if you have additional insights on this, please let us know.

[the remark this answers is not in the document]
Comment (AlonaKaplan) [reply pkg/virt-controller/services/template.go:1280]: >  I can confirm it works without it on our e2e tests.

And the tests have `!customPolicyDisabled`? 
@jean-edouard can you please confirm the if we see passt e2e tests passing it measn the customSELinux for passt may be removed?

[the remark this answers is not in the document]
Comment (jean-edouard) [reply pkg/virt-controller/services/template.go:1280]: > > I can confirm it works without it on our e2e tests.
> 
> And the tests have `!customPolicyDisabled`? @jean-edouard can you please confirm that if we see passt e2e tests passing it means the customSELinux for passt may be removed?

Yes, the necessary permissions were added to (container-selinux)[https://github.com/containers/container-selinux] a while back.

**Comment by jean-edouard** [comment]
> @xpivarc @jean-edouard Passt was our last user of the `virt_launcher.process` custom SELinux type, what about if we just remove it completely now?

Sounds good to me.

**Review by jean-edouard — APPROVED** [review]

File: pkg/virt-controller/services/template.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sbrivio-rh) [reply pkg/virt-controller/services/template.go:1280]: Ah, yes, https://github.com/containers/container-selinux/pull/196 and perhaps others.

But at the same time, libvirt's own SELinux policy now ships with passt [support](https://github.com/fedora-selinux/selinux-policy/blob/d9f4a2bbeb91fd95d0c35a90936efb9ea99d2455/policy/modules/contrib/virt.te#L1227) using interfaces exported by passt's SELinux policy. If you happen to use those, I guess things should work.

**Comment by dhiller** [comment]
:stop_sign: please refrain from manual retesting.

:warning: this PR is being held to stop automatic retesting. Quarantine PR https://github.com/kubevirt/kubevirt/pull/11244 is given priority.

/hold

**Pull request merged**